### PR TITLE
feat!: update dependency to eslint v9 @W-17445058

### PR DESCRIPTION
**Pull request by ravijayaramappa**
BREAKING CHANGE: dropping support for ESLint v7 and v8. Supported option is ESLint v9 only. Use older versions of this plugin to use ESLint v7 and v8.

fixes https://github.com/salesforce/eslint-plugin-lwc/issues/150

### CODE REVIEW

File: test/lib/rules/no-node-env-ssr.js [hunk 1]
```diff
@@ -175,29 +175,6 @@ testRule('no-node-env-in-ssr', {
                 },
             ],
         },
-        {
```
Comment (ravijayaramappa) [inline test/lib/rules/no-node-env-ssr.js:178]: Duplicate test

File: test/lib/rules/ssr/no-node-env.js [hunk 2]
```diff
@@ -175,29 +175,6 @@ testRule('ssr/no-node-env', {
                 },
             ],
         },
-        {
```
Comment (ravijayaramappa) [inline test/lib/rules/ssr/no-node-env.js:178]: Duplicate test

File: test/lib/rules/valid-wire.js [hunk 3]
```diff
@@ -66,17 +66,6 @@ testRule('valid-wire', {
             code: `import { wire } from 'lwc';
             import getFoo from 'adapter';
 
-            class Test {
```
Comment (ravijayaramappa) [inline test/lib/rules/valid-wire.js:69]: Duplicate tests here as well

File: lib/rules/no-async-operation.js [hunk 4]
```diff
@@ -26,10 +27,11 @@ module.exports = {
     },
 
     create(context) {
+        const sourceCode = getSourceCode(context);
         return {
             CallExpression(node) {
                 const { callee } = node;
-                const scope = context.getScope();
+                const scope = sourceCode.getScope ? sourceCode.getScope(node) : context.getScope();
```
Comment (wjhsf) [inline lib/rules/no-async-operation.js:34]:
```suggestion
                const scope = getScope(context, node);
```
Since we've got the helper for it. 🤷

**Review by wjhsf — APPROVED** [review]
What is it about these changes that breaks eslint v7/v8?

**Comment by ravijayaramappa** [comment]
@wjhsf Decided to drop support for v7 and v8 for the following reasons:
1. Reduce tech debt: ESLint v8 is [EOL](https://eslint.org/blog/2024/09/eslint-v8-eol-version-support/), so we keep our eslint tooling updated with the latest and no suffer from supporting older versions.
2. Having v8 and v9 will be a maintenance burden for this repo. Since the config format change, some of the tests need two different test configs.

**Pull request merged**